src/redux/saga/employer.ts

```diff
@@ -1,19 +1,22 @@
-import { put, all, takeLatest, select } from 'redux-saga/effects';
-import axios from 'axios';
+import { put, takeLatest, select } from 'redux-saga/effects'
+import axios from 'axios'
+import { AppState } from '../types'
 
-import { registerEmployerSuccess } from '../../redux/actions/employer';
+import { registerEmployerSuccess } from '../../redux/actions/employer'
 
-function* registerEmployerSaga(credentials: Credential) {
+const credentials = (state: AppState) => state.employer.credentials
+function* registerEmployerSaga() {
+  const credentialData = yield select(credentials)
   try {
-    const req = yield axios.post('/employer', { credentials });
-    yield put(registerEmployerSuccess(req.data));
+    const req = yield axios.post('/employer', { credentialData })
```

Comment:
Here is when the shape of the request body is not correct. Should be:
` axios.post('/employer', { credential: credentialData })`

With `credentialData` should have this shape: 
`{ email: abc, password: abc }`

Reply:
Ok.